airlock/static/assets/file_browser/tree.css

```diff
@@ -163,7 +163,9 @@
   --icon-url: url("/static/icons/attach_file.svg");
 }
 
-[data-output-checker][data-request]:not([data-author]) .tree__file.output {
+[data-request][data-request-independent] .tree__file.user_incomplete,
```

Comment:
I think we can maybe get rid of the data-request and data-request-independant attributes altogether?

The logic of what/whether a user can see a thing is done server side, all the css needs is the presence of user_* or request_* classes on the tree node?

Reply:
Tried, didn't work (although I think maybe data-request could go)
The html_classes are added to a PathItem, based on `request_status` and `user_request_status`. (confusingly named, because `request_status` here is the combined `RequestFileReviewStatus` rather than the status of the release_request.  A PathItem can have both of these, and which one you actually show to a reviewer is dependent on the status of the request, and the PathItem doesn't know anything about that.

Although, thinking about it, maybe we can change the logic in the tree building before the PathItem is created, so it only gets one of user_request_status and request_status depending on what status the release request is in.  That would move the logic out of the template without PathItem having to know about requests.  I will investigate.

Reply:
Right - I think that was the conclusion I came to when talking it though with @tomodwyer, that we should have one status class, not two. Originially, I had though we might show both statuses in the tree, which is why both are added.

However, which we show would depend on the phase of the request, which we don't have yet (working on it!), so may need to wait till that lands.

Reply:
OK, I have refactored the logic into the file browser API (haven't tackled turning it into a single status class yet), and removed it from the template, which is all working. However, I think I may have implemented a visibility bit that isn't quite there yet.
I ended up taking many screenshots, so I've put them in a [slideshow](https://docs.google.com/presentation/d/1w7r5mZGxIRO9qS-H6BHT6q6hfmevBRAFX_Ce27BDxwc/edit?usp=sharing) because it seemed the easiest way to walk through and check what things look like at each stage